### PR TITLE
Adding support for nodejs16 and nodejs18

### PR DESCRIPTION
**Pull request by gligorkot**
## What did you implement:

Adding support for Nodejs 18

## How did you implement it:

Following a template that others have followed

## How can we verify it:

Use it to deploy a nodejs18 function app to Azure.

## Todos:

_**Note: Run `npm run test:ci` to run all validation checks on proposed changes**_

- [x] Ensure there are no lint errors.  
       **Validate via `npm run lint`**  
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [x] Write tests and confirm existing functionality is not broken.  
       **Validate via `npm test`**
- [x] Write documentation
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

**_Is this ready for review?:_** YES
**_Is it a breaking change?:_** NO

### CODE REVIEW

File: src/armTemplates/resources/functionApp.test.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ describe("Function App Resource", () => {
         const config = getConfig(FunctionAppOS.LINUX, Runtime.PYTHON36);
 
         const resource = new FunctionAppResource();
-
```
Comment (medikoo) [inline src/armTemplates/resources/functionApp.test.ts:83]: Let's not introduce unrelated changes

Comment (medikoo) [inline src/armTemplates/resources/functionApp.test.ts:83]: Let's not introduce unrelated changes

File: package-lock.json [hunk 2]
```diff
@@ -5,7 +5,8 @@
   "requires": true,
   "packages": {
     "": {
-      "version": "2.1.0",
```
Comment (medikoo) [inline package-lock.json:8]: Let's not introduce changes to this file

**Review by medikoo — CHANGES_REQUESTED** [review]

File: src/armTemplates/resources/functionApp.test.ts (hunk 1 above)
Comment (gligorkot) [reply src/armTemplates/resources/functionApp.test.ts]: Done

File: package-lock.json (hunk 2 above)
Comment (gligorkot) [reply package-lock.json]: Done

**Comment by gligorkot** [comment]
@medikoo this is now updated and ready for re-review

**Review by medikoo — APPROVED** [review]
Thank you 👍

**Comment by gligorkot** [comment]
@medikoo any chance we can get a new release tagged for these changes you've just merged please? 🙏

**Comment by medikoo** [comment]
@gligorkot Currently validating this repository is broken, and i think we should fix it before pursuing any release.

I've created an issue describing what needs to be done: https://github.com/serverless/serverless-azure-functions/issues/645
